beforematch-security-and-privacy-assessment.md

```diff
@@ -0,0 +1,167 @@
+
+### 2.4. How does this specification deal with sensitive information?
+
+The search queries present in the ScrollToTextFragment and find-in-page features may be sensitive information, and are considered in the "Privacy Considerations" section.
+
+### 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?
+
+This feature does not store any new data in the browser.
+
+### 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?
+
+No information from the underlying platform is exposed by this feature.
+
+### 2.7. Does this specification allow an origin access to sensors on a user's device
+
+No.
+
+### 2.8. What data does this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
+
+This feature does not expose any information about an origin's state and does not enable sending or receving of any data to or from a network connection.
+
+### 2.9. Does this specification enable new script execution/loading mechanisms?
+
+No.
+
+### 2.10. Does this specification allow an origin to access other devices?
+
+No.
+
+### 2.11. Does this specification allow an origin some measure of control over a user agent's native UI?
+
+This feature allows the page to respond to find-in-page, which is part of the user agent's native UI. However, this feature does not allow control over find-in-page.
+
+### 2.12. What temporary identifiers might this specification create or expose to the web?
+
+None.
+
+### 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?
+
+I'm assuming first-party and third-party contexts here refers to cross origin iframes.
+* In the ElementFragment case, the scrolling, fragments, and beforematch events are fully insulated from each other when they occur in different cross origin iframes.
+* In the ScrollToTextFragment case, text inside of a cross origin iframe will not be searched for or scrolled to. Therefore, no beforematch event will be emitted.
+* In the find-in-page case, content in cross origin iframes can be searched for and scrolled to. I'm not sure if and where we should fire beforematch in this case.
+
+### 2.14. How does this specification work in the context of a user agent's Private Browsing or "incognito" mode?
+
+This feature will work the same regardless of incognito mode being enabled or disabled.
+
+### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?
+
+I included a Privacy Considerations section further down this document. I did not include a Security Considerations section because I do not believe this feature exposes any security vulnerabilities.
```

Comment:
FWIW, when we write the spec for this feature, I think we should include a Privacy section in there

Reply:
Should I keep this one? I'd be hesitant to remove it and say it will just be included later in a spec...

Reply:
You can keep this one. It's just this questionnaire is usually done when there is already a spec draft, and it's asking whether the spec draft as this section :) When we have the spec draft, we definitely will have this section, but it's good to keep it here for now as well.

Reply:
Sounds good, thanks for letting me know